torchhd/datasets/dataset.py

```diff
@@ -7,8 +7,202 @@
+        "PageBlocks",
+        "Parkinsons",
+        "Pendigits",
+        "Pima",
+        "PittsburgBridgesMaterial",
+        "PittsburgBridgesRelL",
+        "PittsburgBridgesSpan",
+        "PittsburgBridgesTOrD",
+        "PittsburgBridgesType",
+        "Planning",
+        "PlantMargin",
+        "PlantShape",
+        "PlantTexture",
+        "PostOperative",
+        "PrimaryTumor",
+        "Ringnorm",
+        "Seeds",
+        "Semeion",
+        "Soybean",
+        "Spambase",
+        "Spect",
+        "Spectf",
+        "StatlogAustralianCredit",
+        "StatlogGermanCredit",
+        "StatlogHeart",
+        "StatlogImage",
+        "StatlogLandsat",
+        "StatlogShuttle",
+        "StatlogVehicle",
+        "SteelPlates",
+        "SyntheticControl",
+        "Teaching",
+        "Thyroid",
+        "TicTacToe",
+        "Titanic",
+        "Trains",
+        "Twonorm",
+        "VertebralColumn2Clases",
+        "VertebralColumn3Clases",
+        "WallFollowing",
+        "Waveform",
+        "WaveformNoise",
+        "Wine",
+        "WineQualityRed",
+        "WineQualityWhite",
+        "Yeast",
+        "Zoo",
+    ]
+
+    # Specify namedtuple format
+    DATASET = namedtuple("Dataset", ["name", "train", "test"])
+
+    def __init__(
+        self,
+        root: str,
+        download: bool,
+    ):
+        super(UCIClassificationBenchmark, self).__init__()
+        self.root = root
+        self.download = download
+        self.statistics = {key: [0.0, 0.0] for key in self.UCI_DATASET_COLLECTION}
+
+    # DK there could be better a way to make this generator within the class but I did not find it
+    def datasets(self):
+        # For all datasets in the collection
+        for i in range(len(self.UCI_DATASET_COLLECTION)):
+            # Fetch the current dataset
+            dataset = getattr(torchhd.datasets, self.UCI_DATASET_COLLECTION[i])
+
+            # If no separate test dataset available - do 4-fold cross-validation
+            if hasattr(dataset, "num_folds"):
+                for fold_id in range(dataset.num_folds):
+                    # Set test and train datasets for the current fold
+                    train_ds = dataset(
+                        self.root, train=True, download=self.download, fold=fold_id
+                    )
+                    test_ds = dataset(
+                        self.root, train=False, download=False, fold=fold_id
+                    )
+                    yield self.DATASET(
+                        self.UCI_DATASET_COLLECTION[i], train_ds, test_ds
+                    )
+            # Case of avaiable test set
+            else:
+                # Set test and train datasets
+                train_ds = dataset(self.root, train=True, download=self.download)
+                test_ds = dataset(self.root, train=False, download=False)
+                yield self.DATASET(self.UCI_DATASET_COLLECTION[i], train_ds, test_ds)
+
+    def report(self, dataset, accuracy):
```

Comment:
I think the only thing we should do in the score function is average over the folds, I don't think we should average over the repeats. So if you did 4 repetitions I think the dict should return a list of 4 accuracy values per dataset. I know this makes the implementation more difficult but it would allow the user to get variance and percentile stats for their method.

Reply:
Done

Reply:
Great! thanks :)